public/app/plugins/panel/heatmap/utils.ts

```diff
@@ -280,17 +302,31 @@ export function prepConfig(opts: PrepConfigOpts) {
                 }
               }
 
+              // For pre-bucketed data with explicit scale, calculate expansion factor from actual bucket spacing
+              // For calculated heatmaps, use the full log base
+              let expansionFactor: number = yExp;
+
+              if (yBucketScale !== undefined) {
+                // Try to infer the bucket factor from the actual data spacing
+                const yValues = u.data[1]?.[1];
+                if (Array.isArray(yValues) && yValues.length >= 2 && typeof yValues[0] === 'number') {
+                  expansionFactor = calculateBucketFactor(yValues, yExp);
+                }
+              }
+
               if (dataRef.current?.yLayout === HeatmapCellLayout.le) {
                 if (!minExpanded) {
-                  scaleMin /= yExp;
+                  scaleMin /= expansionFactor;
                 }
               } else if (dataRef.current?.yLayout === HeatmapCellLayout.ge) {
                 if (!maxExpanded) {
-                  scaleMax *= yExp;
+                  scaleMax *= expansionFactor;
                 }
               } else {
-                scaleMin /= yExp / 2;
-                scaleMax *= yExp / 2;
+                // Unknown layout - expand both directions
+                const factor = Math.sqrt(expansionFactor); // Use sqrt for balanced expansion
+                scaleMin /= factor;
+                scaleMax *= factor;
```

Comment:
Fix unknown layout expansion
Use sqrt(expansionFactor) instead of expansionFactor / 2